src/pointer.cr

```diff
@@ -242,7 +242,7 @@ struct Pointer(T)
     raise ArgumentError.new("Negative count") if count < 0
 
     if self.class == source.class
-      Intrinsics.memcpy(self.as(Void*), source.as(Void*), (count * sizeof(T)).to_u32, 0_u32, false)
+      Intrinsics.memcpy(self.as(Void*), source.as(Void*), count * sizeof(T), 0_u32, false)
```

Comment:
I have no idea why the CI isn't broken on 32bits but isn't this passing 32bit ints to something expecting a `U64`?